src/modm/platform/adc/stm32f3/adc_impl.hpp.in

```diff
@@ -142,31 +143,96 @@ modm::platform::Adc{{ id }}::setLeftAdjustResult(const bool enable)
 }
 
 bool
-modm::platform::Adc{{ id }}::setChannel(	const Channel channel,
-										const SampleTime sampleTime)
+modm::platform::Adc{{ id }}::setChannel(const nonstd::span<const Channel> channels)
+{
+	if(channels.size() >= maxSequenceLength) {
+		return false;
+	}
+
+	// clear SQRx registers
+	ADC{{ id }}->SQR1 = 0;
+	ADC{{ id }}->SQR2 = 0;
+	ADC{{ id }}->SQR3 = 0;
+	ADC{{ id }}->SQR4 = 0;
+
+	for(unsigned int i = 0; i < channels.size(); i++) {
+		if(!setSequenceChannel(i, channels[i])) {
+			return false;
+		}
+	}
+	ADC{{ id }}->SQR1 |= channels.size();
```

Comment:
This would then be something like this using [fold expressions](https://en.cppreference.com/w/cpp/language/fold)?

```cpp
int i{0};
bool success = (setSequenceChannel(i++, channels) && ...); // t
ADC{{ id }}->SQR1 |= sizeof...(channels);
```
You could probably also compute the sequence bit-mask before-hand in this function, and then write them directly as a sequence. That would remove the need for the max 16 function calls, which currently allow random-access writes, which isn't really necessary.

Reply:
fold expressions are (unfortunately) not usable with arrays (or std::span).

I'm thinking if I can find a nice solution which accesses each SQRx register only once at most without generating too much overhead. Do you have a suggestion?

Reply:
Locally compute the register then write it only once? Like you do for the injected sequence already but mod 4.